compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/AggregatedOpInterfaceImpl.cpp

```diff
@@ -225,4 +261,265 @@ OnlineAttentionOp::decomposeOperation(OpBuilder &b) {
+  }
+
+  // Delinearize the k_offset into an offset into the convolution window and
+  // any reduced channels. For an NHWC conv2d, the basis for delinearization
+  // would be [P, Q, C] for a PxQ kernel with C channels.
+  Location nestedLoc =
+      loopNest.loops.back().getBody()->getTerminator()->getLoc();
+  b.setInsertionPointToStart(loopNest.loops.back().getBody());
+
+  SmallVector<OpFoldResult> kBasis;
+  SmallVector<int64_t> mKernelIdx(getInputRank(), -1);
+  for (auto [idx, mPos] : enumerate(getMPos())) {
+    mKernelIdx[mPos] = idx;
+  }
+  for (auto [idx, size] : enumerate(inputSizes)) {
+    if (batchPosSet.contains(idx))
+      continue;
+    if (mPosSet.contains(idx)) {
+      kBasis.push_back(kernelSize[mKernelIdx[idx]]);
+      continue;
+    }
+    kBasis.push_back(size);
+  }
+  OpFoldResult kIndex = kOffset.front();
+  if (tileK) {
+    kIndex = addOfrs(b, nestedLoc, kOffset.front(), ivs.back());
+  }
+  FailureOr<SmallVector<Value>> maybeDelinKOffset = affine::delinearizeIndex(
+      b, nestedLoc, getValueOrCreateConstantIndexOp(b, loc, kIndex),
+      getValueOrCreateConstantIndexOp(b, loc, (kBasis)));
+  if (failed(maybeDelinKOffset)) {
+    return failure();
+  }
+  SmallVector<Value> delinKOffset = maybeDelinKOffset.value();
+  // Split the delinearized offsets into the window offsets (for M offsets)
+  // and the K offsets for the input tensor.
+  SmallVector<Value> windowOffset, inputKOffset;
+  int delinKIdx = 0;
+  for (int i = 0; i < getInputRank(); ++i) {
+    if (batchPosSet.contains(i))
+      continue;
+    if (mPosSet.contains(i)) {
+      windowOffset.push_back(delinKOffset[delinKIdx++]);
+      continue;
+    }
+    inputKOffset.push_back(delinKOffset[delinKIdx++]);
+  }
+
+  // Compute offsets for extract. Start by delinearizing the combined offset
+  // of m_offset and the offset from the tiled loop, using the mBasis. This
+  // will give an index into the delinearized output space of the convolution.
+  Value mArg = tileK ? ivs[ivs.size() - 2] : ivs.back();
+  AffineExpr d0, d1;
+  bindDims(b.getContext(), d0, d1);
+  auto addMap = AffineMap::get(2, 0, {d0 + d1});
+  OpFoldResult linearMOffset = affine::makeComposedFoldedAffineApply(
+      b, nestedLoc, addMap, {mArg, mOffset[0]});
+  FailureOr<SmallVector<Value>> maybeDelinMOffset = affine::delinearizeIndex(
+      b, nestedLoc,
+      getValueOrCreateConstantIndexOp(b, nestedLoc, linearMOffset), mBasis);
+  if (failed(maybeDelinMOffset)) {
+    return failure();
+  }
+  SmallVector<Value> delinMOffset = maybeDelinMOffset.value();
+
+  // Compute the final offsets into the input tensor.
+  SmallVector<OpFoldResult> sliceOffsets(
+      getInputRank(), getAsIndexOpFoldResult(b.getContext(), 0));
+  SmallVector<OpFoldResult> sliceStrides(
+      getInputRank(), getAsIndexOpFoldResult(b.getContext(), 1));
+  SmallVector<OpFoldResult> sliceSizes(
+      getInputRank(), getAsIndexOpFoldResult(b.getContext(), 1));
```

Comment:
nit: `getAsIndexOpFoldResult(b.getContext(), 0) -> b.getIndexAttr(0)`

I think it looks better if we declare variables for them, and it can be reused in the later for loop.

```
OpFoldResult zero = b.getIndexAttr(0);
OpFoldResult one = b.getIndexAttr(1);
SmallVector<OpFoldResult> sliceOffsets(getInputRank(), zero);
// ...
```